### DISCUSSION TITLE
Proper Support for anything using host

### DISCUSSION THREAD

**Discussion by UntrustedTech** [thebaselab/codeapp#519] [Ideas]
So I've been working on upgrading my iPads looks and all, i got me a dongle that i can leave connected to my iPad to use my 1tb micro sd card ( before i used to just connect things via adapters ) and before I wouldn't check how my website looks with code i would usually just open it through files, but once i tried it one time using code it wouldn't host properly, it would just end up being a blank page due to my work being in a external memory. Im not sure if its just my iPad since im also using iPadOS 16 but when trining it on my jailbroken iPad mini with the M1 chip also, i get the same issue and its on iPadOS 14.8

**Comment by bummoblizard** [comment]
Do you mean it's unable to open any file located in an external storage?

**Reply by UntrustedTech** [reply]
No it can open any and im able to edit them, its just when ever i want to run it when the files are in an external source it either crashes or shows a blank screen, but when i transfer the files over to iPad internal storage it works fine, so rn im just trying to diagnose if its a problem with IPadOS 16 - 15 & 14 or if its just a problem with swift and external sources, cause ive noticed similar problems when looking into other apps. _Please Correct me if im wrong_

**Comment by bummoblizard** [comment]
I’ll investigate. Thanks for letting me know.

**Reply by UntrustedTech** [reply]
Small update, found out that when trying to run something from an external drive code is able to edit and see it, but technically it doesn't know where exactly the files is in when trying to run it, i say this cause other apps also have the same issue, the file structure would normally be `\projects\whatevertheprojectfileis\bruh.html` because its in the internal drive of the ipad. Now if i try to read the same file but this time its in a external drive, code and every other app reads them as `...\projects\whatevertheprojectfileis\bruh.html` when its trying to run the file. You get what i mean, I don't think its your app itself, i think its just the way its being implemented and being read off to run the file. Cause in Luna fusion when editing of a external drive, it identifies that there is another source connected and can tell that it is. I do think im wrong on how Xcode works and all but with the things ive been trying i get the same result despite if its a usb drive, ssd, or micro sd card.

**Reply by UntrustedTech** [reply]
Update to the update, ok so im not sure why but now i tried to just open the file off the root of my external drive and it was able to identify that it was there, so when i press run it works, so now i just don't know if before this was just a bug or something, i tried it with the other apps but they still have the same issue as before.

**Reply by UntrustedTech** [reply]
Oh and now the external drive identifies as this... ![image](https://user-images.githubusercontent.com/82400546/179302204-767bc106-03cb-4702-bca5-1398e3bb871a.jpeg) and not what its actually named, and code can see hidden files regardless of having that option off.
